### PR TITLE
split atmos and coupler configs - rebased

### PR DESCRIPTION
**Pull request by LenkaNovak**
## Purpose 

Closes https://github.com/CliMA/ClimaCoupler.jl/issues/417

(rebased from #418 )

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: experiments/AMIP/modular/components/atmosphere/climaatmos_init.jl [hunk 1]
```diff
@@ -21,10 +21,32 @@ struct ClimaAtmosSimulation{P, Y, D, I} <: AtmosModelSimulation
...
+function get_atmos_config(coupler_dict)
+    atmos_file = coupler_dict["atmos_config_file"]
+    # override default or specified configs with coupler arguments, and set the correct atmos config_file
+    if isnothing(atmos_file)
+        @info "Using Atmos default configuration"
+        merge(CA.override_default_config(atmos_file), coupler_dict, Dict("config_file" => atmos_file))
+    else
+        @info "Using Atmos configuration from $atmos_file"
+        merge(
+            CA.override_default_config(joinpath(pkgdir(CA), atmos_file)),
```
Comment (juliasloan25) [inline experiments/AMIP/modular/components/atmosphere/climaatmos_init.jl:39]: Why do we need to call `joinpath` here but not in the `if` branch?

File: experiments/AMIP/modular/components/land/bucket_init.jl [hunk 2]
```diff
@@ -3,6 +3,7 @@ using ClimaCore
 using ClimaLSM
 import ClimaLSM
 import ClimaTimeSteppers as CTS
+import Thermodynamics as TD
```
Comment (juliasloan25) [inline experiments/AMIP/modular/components/land/bucket_init.jl:6]: Do we need this? I don't see `TD` used in this file

**Review by juliasloan25 — APPROVED** [review]
LGTM!

File: experiments/AMIP/modular/components/land/bucket_init.jl (hunk 2 above)
Comment (LenkaNovak) [reply experiments/AMIP/modular/components/land/bucket_init.jl:6]: Ahh, it's used in the `bucket_utils.jl`

File: experiments/AMIP/modular/components/atmosphere/climaatmos_init.jl (hunk 1 above)
Comment (LenkaNovak) [reply experiments/AMIP/modular/components/atmosphere/climaatmos_init.jl:39]: In the if part, `atmos_file = nothing`, so this [override function](https://github.com/CliMA/ClimaAtmos.jl/blob/e2268a7b3c9920df5afa10484672f13f5a6e6d81/src/solver/yaml_helper.jl#L63) defaults to `CA.default_config_dict()`. But you're right that this is not obvious. I'll chage this to `CA.default_config_dict()`.

**Comment by LenkaNovak** [comment]
Thank you for the review, @juliasloan25 !

**Pull request merged**